app/controllers/action_mailbox/ingresses/amazon_ses/s3_download.rb

```diff
@@ -13,10 +13,27 @@ def initialize(bucket:, key:, region:)
         end
 
         def content
-          Aws::S3::Client.new(region: region, access_key_id: access_key_id, secret_access_key: secret_access_key)
-                         .get_object(key: key, bucket: bucket)
-                         .body
-                         .string
+          s3_client = Aws::S3::Client.new(
+            region: region,
+            access_key_id: access_key_id,
+            secret_access_key: secret_access_key
+          )
+
+          s3_object = s3_client.get_object(key: key, bucket: bucket)
+
+          if (kms_cmk_id = JSON.parse(s3_object.metadata['x-amz-matdesc'])['kms_cmk_id'])
+            s3_encryption_client = Aws::S3::EncryptionV2::Client.new(
+              client: s3_client,
+              kms_key_id: kms_cmk_id,
+              key_wrap_schema: :kms_context,
+              content_encryption_schema: :aes_gcm_no_padding,
+              security_profile: :v2_and_legacy
```

Comment:
Let's add an inline comment here explaining why we need this 

```suggestion
              security_profile: :v2_and_legacy # SES uses ...
```